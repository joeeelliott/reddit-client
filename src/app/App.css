/* APP COMPONENT */


.App {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  overflow: hidden;
  background-color: rgba(120, 160, 187, 0.2);
}

.App_loading-wrapper {
  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  background-color: #242f3f;
  display: flex; 
  justify-content: center;
  align-items: center;
  /* display: none; */
}

.loader {
  display: inline-block;
  width: 30px;
  height: 30px;
  position: relative;
  border: 4px solid #fff;
  animation: loader 2s infinite ease; 
}

.loader-inner {
  vertical-align: top; 
  display: inline-block;
  width: 100%;
  background-color: #fff;
  animation: loader-inner 2s infinite ease-in; 
}

@keyframes loader {
  0% {
    transform: rotate(0deg);
  }

  25% {
    transform: rotate(180deg);
  }

  50% {
    transform: rotate(180deg);
  }

  75% {
    transform: rotate(360deg); 
  }

  100% {
    transform: rotate(360deg);
  }
}

@keyframes loader-inner {
  0% {
    height: 0%;
  }

  25% {
    height: 0%;
  }

  50% {
    height: 100%;
  }

  75% {
    height: 100%;
  }

  100% {
    height: 0%;
  }
}

/* .App_header-container {
  position: fixed;
  width: 100%;
  top: 0px;
} */

/* HEADER COMPONENT */


.header_header {
  /* border: 1px solid blue; */
  height: 80px;
  background-color: white;
  display: flex;
  flex-direction: column;
  /* align-items: center; */

  justify-content: center;
  /* offset-x | offset-y | blur-radius | color */
  box-shadow: 5px 5px 5px #ddd;

  position: relative; 
  z-index: 100; 

  position: fixed;
  width: 100%;
  top: 0px;
}

.header_toggle-and-title-container {
  /* border: 1px solid blue; */
  display: flex;
  height: 75%;
  position: relative;
}

.header_toggle-btn-container {
  /* border: 1px solid green; */

  position: absolute;
  left: 0px;
  height: 100%;
  width: 10%;
  display: flex;
  justify-content: center;
  align-items: center;

  z-index: 1; /* so event handler works with absolute positioning */
}

.header_logo-title-container {
  /* border: 1px solid red; */
  /* height: 100%; */
  /* width: 80%; */
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  /* margin-right: 20%;  so the logo and title are in the middle of the header, counters the toggle icon 20% width, didn't want toggle icon in absolute positioning  */
}

.header_logo-title-link {
  display: flex;
  align-items: center;
  justify-content: center;
  text-decoration: none;
}

.header_search-icon {
  cursor: pointer;
  /* font-size: 1.5rem !important; */
  transition: all 0.5s ease;  /* part 1 of the burger menu rotation */
}

.header_filter-icon {
  font-size: 0.8rem !important;
  cursor: pointer;
  /* transition: all 0.5s ease; */
   /* part 1 of the burger menu rotation */
}

.header_search-icon-rotate {
  transform: rotate(90deg); /* part 2 of the burger menu rotation */
}

.header_reddit-icon {
  color: rgb(255, 69, 0);
}

.header_font-awesome-icon {
  font-size: 1.5rem;
}

.header_title {
  font-size: 1.5rem;
  color: black;
}

.header_title-span {
  color: grey
}

.header_nav-container {
  /* border: 1px solid green; */
  height: 25%;
  width: 100%;
  display: flex;
  align-content: center;
  justify-content: center;
  /* margin-top: 10px; */
}


/* NAV COMPONENT */


.nav_list {
  /* border: 1px solid yellow; */
  display: flex;
  /* justify-content: center; */
  /* align-items: center; */
  list-style: none;
  height: 100%;
}

.nav_link {
  text-decoration: none;
  margin: 0px 10px;
  /* color: black; */
  height: 100%;
  line-height: 100%;
  color: rgb(150, 150, 150);
  /* border: 1px solid orange; */
}

.nav_link:hover {
  color: black;
}

.nav_saved-article-animation {
  animation: saveClicked 1s;
}

@keyframes saveClicked {
  0% {color:rgb(150, 150, 150);}
  25% {color:rgba(144, 238, 144, 0.5);}
  50% {color:rgba(144, 238, 144, 0.5);}
  75% {color:rgba(144, 238, 144, 0.5);}
  100% {color:rgb(150, 150, 150);}
}


/* SIDENAV COMPONENT */

.sideNav_sideNav {
  position: absolute; 
  left: 0;
  top: 0; 
  height: calc(100vh - 80px);
  margin-top: 80px;
  z-index: -10; 
  font-size: 0.7rem;
  /* width: 170px; */
  width: 0px;
  overflow: hidden;
  background-color: #ddd;
  border-top-right-radius: 20px;
  border-bottom-right-radius: 20px;
}

.sideNav_hide-nav {
  /* border: 1px solid red; */
  /* position: absolute; 
  left: 0;
  top: 0;  */
  /* height: 100vh;  */
  /* margin-top: 80px;
  z-index: -10; 
  font-size: 0.7rem; */
  width: 0px;
  /* overflow: hidden;
  background-color: #ddd; */
  transition: width 0.4s;
}

.sideNav_show-nav {
  /* border: 1px solid red; */
  /* position: absolute; 
  left: 0;
  top: 0;  */
  /* height: calc(100vh - 80px); */
  /* margin-top: 80px;
  z-index: -10; 
  font-size: 0.7rem; */
  width: 170px;
  /* overflow: hidden; 
  background-color: #ddd; */
  transition: width 0.5s;
  /* border-top-right-radius: 20px;
  border-bottom-right-radius: 20px; */
}

.sideNav_hide-nav label, .sideNav_show-nav label {
  display: block; 
  margin: 10px; 
}

.sideNav_hide-nav input, select, .sideNav_show-nav input, select {
  font-size: 0.7rem;
  margin: 0px 10px;
}

.sideNav_btn-container {
  display: flex;
  justify-content: flex-end;
  margin: 20px 10px; 
}

.sideNav_btn {
  font-size: 0.7rem;
  cursor: pointer;
  border: 1px solid grey; 
  border-radius: 20px;
}

.sideNav_eye-text-container {
  /* border: 1px solid blue; */
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
}

.sideNav_eye-icon-container {
  /* border: 1px solid green; */
  display: flex;
  align-items: center;
}

.sideNav_eye-icon {
  font-size: 1rem;
  margin-right: 20px;
  margin-left: 10px;
  cursor: pointer;
}

.sideNav_text-container {
  /* border: 1px solid red; */
}

.sideNav_eye-icon-hover-text {
  font-size: 0.7rem;
  color: rgba(0, 0, 0, 0.5); 
  opacity: 0;
}

.sideNav_eye-icon-hover-text-show {
  animation: showText 2s; 
}

@keyframes showText {
  from {opacity: 0;}
  to {opacity: 1.0;}
}





/* ARTICLE COMPONENT */

.article_container {
  /* border: black 1px solid; */
  border-radius: 15px;
  margin: 100px 13%;  /* make % smaller the wider you want the container. 100px top to fall below the header, 100 from bottom to be able to see the ellipsis dropdown on bottom article */
}

.article_saved-articles-none-saved {
  /* border: 1px solid black; */
  width: 100%; 
  height: calc(100vh - 220px);  /* 100vh take the footer, header, and margins/padding inbetween - didnt actually work out just kept doing trial and error and saw how it changed on the page */ 
  display: flex;
  flex-direction: column; 
  /* align-items: center; */
  justify-content: center;
}

.article_saved-articles-none-saved * {
  text-align: center;
  margin: 10px 0px;
}

.article_saved-articles {
  width: 100%; 
  height: calc(100vh - 220px);  /* 100vh take the footer, header, and margins/padding inbetween - didnt actually work out just kept doing trial and error and saw how it changed on the page */ 
}

.article_saved-articles-2plus {
  width: 100%;
  height: 100%;
}

.article_outer-div {
  position: relative;
}

.article_outer-container {
  /* border: red 1px solid !important; */
  width: 100%;
  display: flex;
  padding: 5px 0px;
  margin-bottom: 20px;
  background-color: #FFFFFF;
  border: 1px solid #ccc;
  border-radius: 5px;
}

.article_outer-container-saved-on-render {
  /* border: red 1px solid !important; */
  width: 100%;
  height: 100%;
  display: flex;
  padding: 5px 0px;
  margin-bottom: 20px;
  background-color: #FFFFFF;
  border: 1px solid #ccc;
  border-radius: 5px;
}

.article_outer-container-hide {
  display: none;
}

.article_score-container {
  /* border: 1px solid green; */
  flex: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  padding-top: 10px;
}

.article_score {
  /* border: 1px solid lightblue; */
  margin: 10px 0px;
  font-weight: 600;
  display: flex;
  justify-content: center;
  align-items: center;
}

.article_score-icon, .article_comments-icon, .article_share-icon, .article_ellipsis-icon {
  color:rgb(150, 150, 150);
  cursor: pointer; 
}

.article_score-up-icon-clicked {
  color: green; 
}

.article_score-down-icon-clicked {
  color: red;
}

.article_main-content-container {
  /* border: 1px solid darkblue; */
  flex: 12;
  display: flex;
  flex-direction: column;
}

.article_details-top-container {
  /* border: 1px solid green; */
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.article_post-details {
  /* border: 1px solid purple; */
  display: flex;
  align-items: center;
  padding-left: 10px;
}

.article_api-data {
  /* border: 1px solid red; */
  color:rgb(150, 150, 150);
  font-size: 0.8rem;
  text-align: center;
}

.strong {
  color: black !important; 
  font-weight: 600;
  margin-right: 5px;
}

.article_join-btn {
  background-color:rgb(31, 144, 231);
  color: #fff;
  border-radius: 20px;
  padding: 3px 5px;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 10px;
  cursor: pointer;
}

.article_join-btn-text {
  margin: 0px 5px;
  font-size: 0.8rem;
}

.article_title-bottom-details-img-container {
  /* border: 1px solid red; */
  display: flex; 
  justify-content: space-between;
  height: 100%;
}

.article_title-bottom-details-container {
  /* border: 1px solid blue; */
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.article_title-container {
  /* border: 1px solid red; */
  padding-left: 10px;
}

.article_title {
  /* border: 1px solid green; */
  font-size: 1rem;
  font-weight: 500;
}

.article_bottom-details-container {
  /* border: 2px solid yellow; */
  display: flex;
  flex-direction: row;
  align-items: center;
  font-size: 0.8rem;
  margin-top: 10px;
  padding-left: 10px;
}

.article_comments-icon-container, .article_share-icon-container {
  display: flex;
  align-items: center;
  padding: 0px 4px;
  cursor: pointer;
}

.article_share-icon-container {
  margin-left: 25px;
}

.article_comments-icon-container:hover, .article_share-icon-container:hover {
  background-color:rgba(120, 160, 187, 0.2)
}

.article_comments-icon {
  margin-right: 5px;
}

.article_share-icon {
  margin-right: 5px;
}

.article_ellipsis-container {
  margin-left: 25px;
  position: relative;
  /* border: 1px solid black; */
  /* background-color: green; */
  padding: 0px 4px;
}

.article_ellipsis-container:hover {
  background-color:rgba(120, 160, 187, 0.2)
}

.article_ellipsis-icon {
  /* margin-left: 25px; */
  font-weight: normal;
  cursor: pointer;
}

.article_ellipsis-dropdown {
  /* border: 1px solid green; */
  /* height: 60px;
  width: 10px; */
  position: absolute;
  left: 0px;
  bottom: -80px;
  background-color: rgb(248, 248, 240); 
  /* background-color: red; */
  display: none;
  z-index: 55;
}

.article_ellipsis-dropdown-show {
  display: block !important; 
}

.article_save-container, .article_hide-container, .article_report-container {
  /* border: 1px solid red; */
  width: 100px;
  height: 25px;
  display: flex;
  flex-direction: row;
  /* justify-content: space-between; */
  cursor: pointer; 
  color: rgba(0, 0, 0, 0.5); 
  background-color: rgb(248, 248, 240); 
}

.article_save-container-clicked, .article_hide-container-clicked, .article_report-container-clicked {
  width: 100px;
  height: 25px;
  display: flex;
  flex-direction: row;
  /* justify-content: space-between; */
  cursor: pointer; 
  color: rgba(0, 0, 0, 0.5);
  background-color: rgba(144, 238, 144, 0.5); 
}

.article_report-container-clicked {
  background-color: rgba(255, 0, 0, 0.8) !important; 
}

.article_save-container:hover, .article_hide-container:hover, .article_report-container:hover {
  background-color: rgb(230, 230, 230); 
  color: #000; 
}

.article_save-container-clicked:hover, .article_hide-container-clicked:hover {
  background-color: rgba(144, 238, 144, 0.5);  
  color: #000; 
}

.article_report-container-clicked:hover {
  color: #000;
}

.article_save-icon-container, .article_hide-icon-container, .article_report-icon-container {
  /* border: 1px solid blue;  */
  width: 40%;
  /* width: 80px;
  height: 25px;
  display: flex;
  flex-direction: row; */
  display: flex;
  align-items: center;
  justify-content: center;
  /* text-align: center; */
}

.article_save-icon, .article_hide-icon, .article_report-icon {
  /* margin-right: 5px; */
  /* flex: 2; */
  /* height: 100%; */
  /* font-size: 0.5rem; */
  font-size: 0.9rem;
  width: 30px;
  /* font-weight: 0.8rem; */
}

.article_save-text, .article_hide-text, .article_report-text {
  /* border: 1px solid black; */
  align-self: center;
  display: flex;
  /* justify-content: center; */  
  /* font-size: 0.9rem; */
}

.article_save-text, .article_hide-text, .article_report-text {
  /* flex: 1; */
  width: 50px;
}

.article_report-text:hover {
  /* background-color: rgb(230, 230, 230); 
  color: #000;  */
}

.article_save-icon-container {
  /* border: 1px solid red; */
}

.article_save-icon {
  /* display: inline-block; */
  /* border: 1px solid black; */
}

.article_hide-icon-container {
  /* border: 1px solid blue; */
}

.article_hide-icon {
  /* border: 1px solid black; */
  /* font-size: 1rem; */
}

.article_report-icon-container {
  /* border: 1px solid yellow; */
}

.article_report-icon {
  /* border: 1px solid black; */
}

.article_img-container {
  /* border: 2px solid orange; */
  padding: 5px;
  display: flex;
  align-items: center;
  position: relative;
}

.article_article-img {
  /* border: 1px solid rgb(0, 121, 211); */
  cursor: pointer;
}

.article_article-img-clicked {
  /* border: 1px solid rgb(0, 121, 211); */
  border-radius: 5px;
}

.article_img-modal {
  position: absolute; 
  border: 1px solid rgba(150, 150, 150);
  border-radius: 5px;
  height: auto;
  width: auto;
  top: 0px;
  right: 0px;
  padding: 20%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(233, 233, 233, 0.9);
  z-index: 50;
}

.article_img-modal-close-icon {
  position: absolute;
  top: 3%;
  right: 3%;
  color: rgba(150, 150, 150);
  cursor: pointer;
}

.article_img-modal-close-icon:hover {
  color: red;
}

.article_report-modal-hidden {
  position: fixed; 
  top: 0;
  left: 0;
  height: 100vh;
  width: 100vw;
  z-index: 100; 
  background-color:rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
  /* display: none; */
  visibility: hidden;  
}

.article_report-modal {
  position: fixed; 
  top: 0;
  left: 0;
  height: 100vh;
  width: 100vw;
  z-index: 100; 
  background-color:rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
  /* display: none; */
  /* visibility: hidden;  */
}

.article_report-modal-content {
  height: 50%;
  width: 50%;
  background-color: rgba(255, 255, 255, 0.9); 
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  text-align: center;
  border-radius: 2%;
  padding: 20px 0px;
}

.article_report-modal-report-icon {
  font-size: 2rem;
  color: red;
}

.article_report-modal-btn-container button {
  padding: 2px;
  background-color: rgba(230, 230, 230);
  border: 1px solid grey;
  border-radius: 5px;
}

.article_report-modal-btn-container button:hover {
  background-color: rgba(0, 0, 0);
  color: #fff;
}


/* FOOTER COMPONENT */


.footer {
  /* position: fixed; */
  /* bottom: 0px; */
  /* border: 1px solid black; */
  width: 100%;
  height: 5%;
  box-shadow: 0px -15px 20px #ddd;  
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #fff;
}

.footer h1 {
  font-size: 0.8rem;
  margin: 0px 2px;
  /* border: 1px solid black; */
}

.footer p {
  font-size: 0.8rem;
  margin: 0px 2px;
}



/* NOTE TO SELF ----> below is how you toggle rotate on each click  */
/* #shape  { width: 100px; 
  height: 200px; 
  background:#000;
  -moz-transition: all 1s ease;      <------- transition property
  -webkit-transition: all 1s ease;
  -o-transition: all 1s ease;
   transition: all 1s ease;
 }

.rotate {-moz-transform: rotate(90deg);   <------ rotate property on 
 -webkit-transform:rotate(90deg);                 a class that will be 
 -ms-transform:rotate(90deg);                     toggled 
 -o-transform:rotate(90deg);
 } */


 
/*       below is example of an infinite rotation animation 
@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
} */

/* @keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
} */

