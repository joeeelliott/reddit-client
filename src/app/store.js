import { configureStore } from '@reduxjs/toolkit'; 

import sideNavSlice from '../features/sideNavSlice'; 
import articleSliceReducer from '../features/articleSlice'; 

export const store = configureStore({
  reducer: {
    sideNav: sideNavSlice,
    articles: articleSliceReducer,
  }
});