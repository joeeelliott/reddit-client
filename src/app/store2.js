// import { configureStore } from '@reduxjs/toolkit'; 

// import sideNavSlice from '../features/sideNavSlice'; 

// export const store = configureStore({
//   reducer: {
//     sideNav: sideNavSlice,
//   }
// });

import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import reducer from '../features/sideNavSlice'; 

const store = state => createStore(reducer, state, applyMiddleware(thunk)); 

export default store; 