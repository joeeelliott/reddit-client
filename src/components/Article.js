import React from 'react';

import Score from './Score'; 
import TopRow from './TopRow'; 
import Title from './Title';
import CommentsIcon from './CommentsIcon';
import ShareIcon from './ShareIcon';
import Ellipsis from './Ellipsis';
import Image from './Image'; 
import ReportModal from './ReportModal';

const Article = ({ id, score, author, created, title, numComments, saved, thumbnail, articles, articleType, scoredUp, scoredDown, hidden, reported, imgClicked }) => {

  return (
    <div className={hidden ? "article_outer-container-hide" : "article_outer-container"} id={id}>
      <Score id={id} articleType={articleType} score={score} currentArticles={articles} scoredUp={scoredUp} scoredDown={scoredDown} />
      <div className="article_main-content-container">
        <TopRow author={author} created={created} /> 

        <div className="article_title-bottom-details-img-container">
          <div className="article_title-bottom-details-container">
            <Title title={title} />
            <div className="article_bottom-details-container">
              <CommentsIcon numComments={numComments} />
              <ShareIcon />
              <Ellipsis id={id} saved={saved} hidden={hidden} reported={reported} articles={articles} articleType={articleType} />
            </div>
          </div>

          {thumbnail.url.includes('https') && <Image id={id} thumbnail={thumbnail} imgClicked={imgClicked} articleType={articleType} />}
        </div>
      </div>

      {reported && <ReportModal reported={reported} />}
    </div>
  )
}

export default Article;