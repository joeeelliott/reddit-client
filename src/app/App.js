import './App.css';

import Header from '../components/Header'; 
import Article from '../components/Article';
import Footer from '../components/Footer'; 

function App() {
  // <script src="jquery-2.1.4.js"></script>

  return (
    <div data-test="App">
      <div className="App_loading-wrapper">
        <span class="loader"><span class="loader-inner"></span></span>
      </div>
      
      <Header />
      <div className="article_container">
        <Article />
      </div>
      <Footer />

      {/* <script>
        $(window).on('load', function() {
          $('.loader-wrapper').fadeOut('slow')
        });
      </script> */}  
      
      {/* {need to find a way to load this in on window load - maybe in article slice will pop as no display for now but thats not currently pushed and merged to master yet so will do that later} */}
    </div>
  );
}

export default App;
