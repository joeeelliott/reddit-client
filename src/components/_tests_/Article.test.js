import React from 'react'; 
import { Provider } from 'react-redux';
import Article from '../Article'; 

import { render as rtlRender, screen, fireEvent, cleanup } from '@testing-library/react'; 
import { store } from '../../app/store';

const render = component => rtlRender(
  <Provider store={store}>   
    {component}
  </Provider>
);

describe('Article Component', () => {
  
  let component;

  beforeEach(() => {
    component = render(<Article />)
  });

  describe('renders in correct elements', () => {
    it('should render in one outer div', () => {
      expect(component.getAllByRole('outer-div')).toHaveLength(1);
    });

    it('should render in one comments div', () => {
      expect(component.getAllByRole('left-sided-column')).toHaveLength(1);
    });

    // dont yet have a test for the render of amount of actual articles - cos dont know how many there will be yet until i do the fetch request

    it('should render in one comments total <p> tag', () => {
      expect(component.getAllByRole('comments-total')).toHaveLength(1);
    });

    it('should render in one main-content <div>', () => {
      expect(component.getAllByRole('main-content-div')).toHaveLength(1);
    });

    it('should render in one title element', () => {
      expect(component.getAllByRole('title')).toHaveLength(1);
    });

    it('should render in one article <div>', () => {
      expect(component.getAllByRole('article')).toHaveLength(1);
    });

    it('should render in one article-details <div>', () => {
      expect(component.getAllByRole('article-details-div')).toHaveLength(1);
    });

    it('should render in three article details <p> tags', () => {
      expect(component.getAllByRole('article-detail')).toHaveLength(3);
    });

  });

});