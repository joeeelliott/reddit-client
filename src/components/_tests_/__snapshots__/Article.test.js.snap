// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Component renders in as expected; matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        role="outer-div"
      >
        <div
          class="article_article-div"
          role="article-div"
        >
          <div
            class="article_comments-column"
            role="left-sided-column"
          >
            <p
              class="article_total-comments"
              role="comments-total"
            >
              Data loading...
            </p>
          </div>
          <div
            class="article_main-content-div"
            role="main-content-div"
          >
            <h1
              class="article_title"
              role="title"
            >
              Article Title
            </h1>
            <div
              class="article_article"
              role="article"
            >
              <img
                class="article_article-img"
                src="mockpic.png"
              />
            </div>
            <div
              class="article_article-details-div"
              role="article-details-div"
            >
              <p
                class="article_article-detail"
                role="article-detail"
              >
                Posted By: Joe Elliott
              </p>
              <p
                class="article_article-detail"
                role="article-detail"
              >
                2hrs ago
              </p>
              <p
                class="article_article-detail"
                role="article-detail"
              >
                22.2k comments
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      role="outer-div"
    >
      <div
        class="article_article-div"
        role="article-div"
      >
        <div
          class="article_comments-column"
          role="left-sided-column"
        >
          <p
            class="article_total-comments"
            role="comments-total"
          >
            Data loading...
          </p>
        </div>
        <div
          class="article_main-content-div"
          role="main-content-div"
        >
          <h1
            class="article_title"
            role="title"
          >
            Article Title
          </h1>
          <div
            class="article_article"
            role="article"
          >
            <img
              class="article_article-img"
              src="mockpic.png"
            />
          </div>
          <div
            class="article_article-details-div"
            role="article-details-div"
          >
            <p
              class="article_article-detail"
              role="article-detail"
            >
              Posted By: Joe Elliott
            </p>
            <p
              class="article_article-detail"
              role="article-detail"
            >
              2hrs ago
            </p>
            <p
              class="article_article-detail"
              role="article-detail"
            >
              22.2k comments
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
